### PR TITLE
Plugin server tests CI

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

Stepping on @fuziontech's toes here to speed things up.

With this in, we can merge #6784  and we'll be fully migrated over to the monorepo.

I also see this as a blocker to merging larger plugin server changes like what @tiina303 is working on

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: .github/workflows/ci-frontend.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 name: Frontend CI
 
 on:
-    - pull_request
+    pull_request:
+        paths-ignore:
+            - 'plugin-server/**'
```
Comment (yakkomajuri) [inline .github/workflows/ci-frontend.yml:6]: frontend tests don't need to run on plugin server changes

File: .github/workflows/cypress-component.yml [hunk 2]
```diff
@@ -1,7 +1,9 @@
 name: Cypress Component
 
 on:
-    - pull_request
+    pull_request:
+        paths-ignore:
+            - 'plugin-server/**'
```
Comment (yakkomajuri) [inline .github/workflows/cypress-component.yml:6]: cypress component tests don't need to run on plugin server changes

File: .github/workflows/plugin-server-ci.yml [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
-    - pull_request
+    pull_request:
+        paths:
+            - 'plugin-server/**'
+            - 'ee/models/**'
+            - 'ee/clickhouse/**'
+            - 'ee/migrations/**'
+            - 'posthog/models/**'
+            - 'posthog/migrations/**'
+            - 'posthog/plugins/**'
```
Comment (yakkomajuri) [inline .github/workflows/plugin-server-ci.yml:12]: plugin server tests are currently quite slow. speeding those up is something we should do, but nevertheless they don't need to _always_ run. here's the directory list I came up with of changes that would affect the plugin server

Comment (hazzadous) [reply .github/workflows/plugin-server-ci.yml:12]: Do we have to be concerned about Github required Checks? I put this in for backend but had to revert: https://github.com/PostHog/posthog/pull/6356

Instead using some paths-filter action to make the tests conditional.

Also, I guess changes to docker-compose might be relevant? Also, I think maybe it doesn't matter about posthog/models/** but just the migrations?

Comment (yakkomajuri) [reply .github/workflows/plugin-server-ci.yml:12]: Definitely not here if we'll probably not make these required for now at least.

Yeah fair point about migrations

Comment (yakkomajuri) [reply .github/workflows/plugin-server-ci.yml:10]: Addressed, thanks

**Comment by yakkomajuri** [comment]
Fine to cover them all, no?

**Comment by hazzadous** [comment]
> Fine to cover them all, no?

Ah maybe, not sure how the pattern matching works, but if it gets the ee docker compose then yup

**Review by tiina303 — APPROVED** [review]
lgtm (verified pattern matching works the way we need https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths)

File: tsconfig.json [hunk 4]
```diff
@@ -28,5 +28,5 @@
         "lib": ["dom", "es2019"]
     },
     "include": ["frontend/**/*"],
-    "exclude": ["node_modules/**/*", "staticfiles/**/*", "frontend/dist/**/*"]
+    "exclude": ["node_modules/**/*", "staticfiles/**/*", "frontend/dist/**/*", "plugin-server/**/*"]
```
Comment (fuziontech) [inline tsconfig.json:31]: I imagine we will want to turn on tsconfig for `plugin-server` or maybe the `tsconfig.json` in that directory will take over for that directory?

**Review by fuziontech — APPROVED** [review]

**Pull request merged**